File: src/main/java/org/kane/domain/service/recipe_recource/measure_unit/MeasureUnitService.java
package org.kane.domain.service.recipe_recource.measure_unit;

import org.kane.domain.DTO.entityDTO.diary.recipe_recource.measure_unit.MeasureUnitDTO;

import java.util.List;

public interface MeasureUnitService {
    List<MeasureUnitDTO> findAllByIngredientID(Long productID);

    MeasureUnitDTO createMeasureUnit(String name);

    List<MeasureUnitDTO> getAllUnits();

    List<MeasureUnitDTO> getFreeUnitsByCategoryID(Long categoryID);
}

File: src/app/service/recipe_resource/measure-unit.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Endpoint} from '../../util/endpoint';
import {Observable} from 'rxjs';
import MeasureUnitDTO from '../../DTO/entity_dto/recipe-recource/measure-unit.dto';

@Injectable({
  providedIn: 'root',
})
export default class MeasureUnitService {
  private readonly http = inject(HttpClient);
  private readonly measureUnitAPI = new Endpoint('measureUnit');

  findAllMeasureUnit(id: number): Observable<MeasureUnitDTO[]> {
    return this.http.get<MeasureUnitDTO[]>(
      this.measureUnitAPI.builder()
        .points("all", id.toString())
        .build())
  }

  createMeasureUnit(name: string):Observable<MeasureUnitDTO>{
    return this.http.get<MeasureUnitDTO>(
      this.measureUnitAPI.builder()
        .points("create")
        .addParam("name", name)
        .build())
  }

  findAll():Observable<MeasureUnitDTO[]>{
    return this.http.get<MeasureUnitDTO[]>(
      this.measureUnitAPI.builder()
        .points("all")
        .build())
  }

  findFreeMeasureUnit(id:number):Observable<MeasureUnitDTO[]>{
    return this.http.get<MeasureUnitDTO[]>(
      this.measureUnitAPI.builder()
        .points("free", id.toString())
        .build())
  }
}
